### PR TITLE
Enable SELinux in tests

### PR DESCRIPTION
**Pull request by mrunalp**
This is to check status of tests with SELinux.

Signed-off-by: Mrunal Patel <mrunalp@gmail.com>

### CODE REVIEW

**Comment by TomSweeneyRedHat** [comment]
Changes LGTM, but e2e RHEL is not happy.  Keeping fingers crossed for the others.

**Comment by rhatdan** [comment]
I take it that all of the errors that look like
Oct 18 17:18:24 ip-172-18-5-249.ec2.internal crio[14217]: time="2017-10-18 17:18:24.847115324Z" level=error msg="Container creation error: container_linux.go:267: creating new parent process caused "container_linux.go:1571: running lstat on namespace path \"/proc/0/ns/ipc\" caused \"lstat /proc/0/ns/ipc: no such file or directory\""
Are causing the problems.

**Comment by mrunalp** [comment]
Yes. We typically see those when SELINUX is blocking something.

**Comment by rhatdan** [comment]
But we are seeing nothing in the AVC logs?  Is there a way to get onto these machines?

**Comment by cevich** [comment]
@rhatdan that wouldn't be related to the same thing fixed in #982 (that also removes the setenforce 0)??

If I rebase this PR locally against #972, I can spin up VMs (internally) in openstack all day long, need one or five?  What flavor?

**Comment by mrunalp** [comment]
@rhatdan updated. PTAL.

File: contrib/test/integration/e2e.yml [hunk 1]
```diff
@@ -1,5 +1,8 @@
 ---
 
+- name: Set SELinux label for cri-o binary
+  command: chcon -t container_runtime_t /usr/local/bin/crio
+
```
Comment (cevich) [inline contrib/test/integration/e2e.yml]: This is fine for testing purposes, but really I think this should be set at cri-o build/install time (from the makefile and/or RPM?).  Once that's fixed, I have NP junking 982, but in all cases the above is probably not a change we want to persist.

Comment (cevich) [reply contrib/test/integration/e2e.yml]: (as discussed on IRC)

Comment (mrunalp) [reply contrib/test/integration/e2e.yml]: :+1: The purpose here is to debug the SELinux issues.

**Comment by mrunalp** [comment]
@rhatdan I am getting this:
```
chcon: failed to change context of '/usr/local/bin/crio' to ‘unconfined_u:object_r:container_runtime_t:s0’: Permission denied
```
Could this be because of any reason other than not being root?

**Comment by mrunalp** [comment]
I changed the chcon to match what I have for my docker binary.

**Comment by rhatdan** [comment]
container_runtime_t was the wrong label, should have been container_runtime_exec_t.  Looks good now.

**Comment by mrunalp** [comment]
okay, so looks like the RHEL tests passed :+1: :)

**Comment by cevich** [comment]
Uggg, I could be mistaken, but last week wasn't this working on Fedora and failing on RHEL?

File: contrib/test/integration/e2e.yml [hunk 2]
```diff
@@ -1,5 +1,8 @@
 ---
 
+- name: Set SELinux label for cri-o binary
+  command: chcon system_u:object_r:container_runtime_exec_t:s0 /usr/local/bin/crio
```
Comment (cevich) [inline contrib/test/integration/e2e.yml]: ``when: ansible_distribution == 'RedHat'``

maybe?

Comment (rhatdan) [reply contrib/test/integration/e2e.yml]: This label should work on both platforms.  Are we getting an error when setting it?

Comment (rhatdan) [reply contrib/test/integration/e2e.yml]: It looks like the chcon did not fail?

**Comment by cevich** [comment]
Different way: #1064

**Comment by rhatdan** [comment]
@mrunalp @cevich What do we want ot do with this?

**Comment by cevich** [comment]
IMHO, this way is simpler but  #1064 allows more easier on/off switching but also needs rebasing (I'll take care of that now).  My preference is all tests run with SELinux enabled/enforcing, so however that gets to happen is fine :smile:

**Comment by cevich** [comment]
Note: The (unintended) side-effect of the old way (just going into permissive) means any subsequent tasks or tests will inherit that state (could be surprising).  Because POLA: I wrote  #1064 so no tests/testers are surprised because status of SELinux enforcement is clear and under control.

This PR accomplishes the same thing, by brute-force  - just set enable/enforcing always and let the cards fall where they fall.  It's less explicit, but the simplicity is attractive.   I think the decision comes down to:  

Will it ever be necessary to control this aspect on a per-test basis in PRs or in general?

If not, then this PR is fine, we can close the other one.

**Comment by rhatdan** [comment]
@mrunalp Needs a rebase

**Comment by cevich** [comment]
@mrunalp since the other PR is merged, enabling SELinux should be super-easy.  Just edit ``vars.yml``

**Comment by mrunalp** [comment]
Rebased and updated to use vars.yml.

**Comment by mrunalp** [comment]
@rhatdan Both e2e failed on the same tests but I see no AVCs. We will need to look while the tests are running.

**Comment by rhatdan** [comment]
Looks like this test is failing?
```
# TODO remove the last test skipped once https://github.com/kubernetes-incubator/cri-o/pull/1217 is merged
- name: Buffer the e2e testing command to workaround Ansible YAML folding "feature"
  set_fact:
    e2e_shell_cmd: >
        /usr/bin/go run hack/e2e.go
            --test
            --test_args="-host=https://{{ ansible_default_ipv4.address }}:6443
                        --ginkgo.skip=\[Slow\]|\[Serial\]|\[Disruptive\]|\[Flaky\]|\[Feature:.+\]|PersistentVolumes|\[HPA\]|should.support.building.a.client.with.a.CSR|should.support.inline.execution.and.attach|should.propagate.mounts.to.the.host
                        --report-dir={{ artifacts }}"
            &> {{ artifacts }}/e2e.log
  # Fix vim syntax hilighting: "
```
Is this test attempting to mount something?

**Comment by rhatdan** [comment]
One thing we could do to get all of the AVC's would be to disable the dontaudit rules and see what AVC's are being dontaudited.

semodule -DB
Turns them off
semodule -B
Turns them back on.

**Comment by cevich** [comment]
***Note***: IIRC, any log-generating command (like ausearch) failures (in an origin-ci job) are mostly silent.  I could be wrong, but I think if the file's is missing or a command-line option is bad, you'll get nothing...which in this case doesn't necessarily mean there were no denials :confused: 

looking at the failed jobs...

**Comment by cevich** [comment]
...ahh, okay, it's only half-bad:

For ``e2e_rhel`` (under s3 artifacts) ``generated/avc_denials.log`` contains: ``<no matches>`` so we know the command is at least running.  However, on e2e_fedora, the file us completely empty :frowning: 

I would interpret that as "the command failed", however, I looked and _ALL_ the ``generated/*`` files are empty.  So either every command failed...or there's some kind of collection problem w/in origin-ci, but it only affects Fedora :confused: 

investigating...

**Comment by cevich** [comment]
...I found in the origin-ci console logs where it's failing for Fedora but working for RHEL.  Still not sure why though.  For example, it does ``ssh test-vm 'sudo ausearch -m AVC -m SELINUX_ERR -m USER_AVC 2>&1' || true``.  Fortunately in both cases, the shell is ``set -x``, so in RHEL the command is successful, but on Fedora it shows ``true`` (meaning it failed).  I'll poke around a bit more to see if I can figure this out, otherwise I'll mail Steve about it.

Bottom-line, log-generating commands are working fine on RHEL, but "silently" failing on Fedora. :face_with_head_bandage:

**Comment by cevich** [comment]
(mailed steve, cc'd y'all)

**Comment by rhatdan** [comment]
In order to get the audit logs to be generated the auditd daemon must be running, I wonder if this is causing the issues?

**Comment by cevich** [comment]
@rhatdan That could happen also :frowning_face:   B/c the way collection happens, we lose all logs if the VM (for any reason) isn't reachable via ssh (first e-mail).  I think in this specific case, and probably in general, failure of the lvm collection is breaking subsequent collection (for whatever reason).  Sent y'all an e-mail just now with a proposed simple, though imperfect solution.

**Comment by rhatdan** [comment]
This is a very old PR.  We have this working on podman, tests on native storage.  Is this still necessary here?

**Comment by mrunalp** [comment]
Will look at master.

**Pull request closed without merging**